README.md

```diff
@@ -1,124 +1,106 @@
+> However there are Java APIs and libraries like [RxJava 2][rxjava2] which don't accept `null` values
+> and language-level nullability cannot help with that.
 
->We don't think that Kotlin itself needs `Optional` because it has strong null-safe type system that effectively eliminates need in such a wrapper. However there are APIs and libraries like [RxJava 2][rxjava2] which don't accept `null` values. 
+> We also think that in many cases you can use `sealed class`es to express absent values,
+> however in simple cases like passing `String?` through RxJava stream `Optional` is a more convenient solution.
 
->We also think that in many cases you can use `sealed class`es to express absent values, however in simple cases like passing `String?` through Rx stream `Optional` is a more convenient solution.
+The goal of this implementation is to be convenient to use and fit Kotlin's `null`-safe type system, which resulted in:
 
----
+* Only two functions: `toOptional()` and `toNullable()`.
+  * Mimics Kotlin functions like `toInt()` and `toBoolean()`.
+* `Some` and `None` are declared as **top level types**.
+  * No need to write `Optional.Some` or `Optional.None`.
+* No functions like `map()`, `getOrElse()`, `filter()`, etc.
+  * Use `toNullable()` and Kotlin `stdlib` functions like `let()`, `takeIf()` instead.
 
-The goal of this implementation is to be convenient to use and fit Kotlin's null-safe type system, which resulted in:
+## Usage
 
-* Only two functions (mimics Kotlin std `toInt()`/`toBoolean()`/etc):
-  - `fun <T : Any> T?.toOptional(): Optional<T>`
-  - `fun Optional.toNullable(): T?`
-* `Some` and `None` are declared as **top level types** — no need to write `Optional.Some` or `Optional.None`
-* No functions like `map()`, `getOrElse()`, `filter()`, etc — apply `toNullable()` and use Kotlin sdt functions like `let()`, `takeIf()` and so on.
-
-### Usage
-
-#### Create `Some`
+### Create
 
 ```kotlin
 val some = Some(value)
-```
-#### Use `None`
-
-```kotlin
 val none = None // It's an object!
 ```
 
-#### Convert `T?` to `Optional<T>`
+### Convert
 
 ```kotlin
-// If something is null — you'll get None, otherwise you'll get Some(something).
-val o = something.toOptional()
-```
+// T? → Optional<T>
+// If value is null — you'll get None, otherwise you'll get Some(value).
+val o = value.toOptional()
 
-#### Convert `Optional<T>` to `T?`
-
-```kotlin
-// If Optional is None — you'll get null, otherwise you'll get not null T value.
+// Optional<T> → T?
+// If optional is None — you'll get null, otherwise you'll get non-null T value.
 val t = optional.toNullable()
 ```
 
-#### Get `value` or fall back to something else if `None` (`getOrElse()`)
+### Leverage Kotlin Features
+
+#### Fallback from `None` (like `java.util.Optional.getOrElse()`)
 
 ```kotlin
 val f = optional.toNullable() ?: "fallback"
 ```
-#### Check if `Optional` is `Some` or `None`
+#### Use [Smart Cast](http://kotlinlang.org/docs/reference/typecasts.html#smart-casts)
```

Comment:
Let's not use word `Use`? It reads like we're forcing users to use these features :)

Reply:
Yeah I understand, I think it's fine in previous headers, but kinda odd here

"Interop with Java" doesn't have verb and looks fine :)